### PR TITLE
Project part1

### PR DESCRIPTION
**Pull request by s-karpovich**
Project part1

### CODE REVIEW

File: web-module/src/test/java/by/tut/mdcatalog/project2/web/controller/UserControllerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    @Test
+    public void shouldRedirectAdminToLoginErrorPageOnLoginWithWrongData() throws Exception {
+        mvc.perform(post("/login")
+                .param("username", "admin@email.com")
+                .param("password", "wrongpassword"))
+                .andExpect(status().is3xxRedirection())
+                .andExpect(redirectedUrl("/login?error"));
+    }
+
+    @WithMockUser(authorities = "ADMINISTRATOR")
```
Comment (artem-perevoznikov) [inline web-module/src/test/java/by/tut/mdcatalog/project2/web/controller/UserControllerIntegrationTest.java]: Need to use constant

File: repository-module/src/main/java/by/tut/mdcatalog/project2/repository/impl/ReviewRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        }
+        return reviewList;
+    }
+
+    @Override
+    public void deleteReviews(Connection connection, int[] ids) {
+        StringBuilder sqlQuery = new StringBuilder("UPDATE review SET deleted=true WHERE id IN (");
+        for (int i = 0; i < ids.length; i++) {
+            if (i != ids.length-1) sqlQuery.append(ids[i]).append(',');
+            else  sqlQuery.append(ids[i]).append(')');
```
[the remark this answers is not in the document]
Comment (s-karpovich) [reply repository-module/src/main/java/by/tut/mdcatalog/project2/repository/impl/ReviewRepositoryImpl.java]: +

File: web-module/src/main/java/by/tut/mdcatalog/project2/web/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import by.tut.mdcatalog.project2.service.UserService;
+import by.tut.mdcatalog.project2.service.model.RoleDTO;
+import by.tut.mdcatalog.project2.service.model.RoleDTOUpdated;
+import by.tut.mdcatalog.project2.service.model.UserDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
```
[the remark this answers is not in the document]
Comment (s-karpovich) [reply web-module/src/main/java/by/tut/mdcatalog/project2/web/controller/UserController.java]: done

File: web-module/src/main/java/by/tut/mdcatalog/project2/web/security/config/ApiSecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        auth.userDetailsService(userDetailsService)
+                .passwordEncoder(passwordEncoder);
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.antMatcher("/api/**")
+                .authorizeRequests()
+                .anyRequest()
+                .hasAuthority("ADMINISTRATOR")
```
[the remark this answers is not in the document]
Comment (s-karpovich) [reply web-module/src/main/java/by/tut/mdcatalog/project2/web/security/config/ApiSecurityConfig.java]: done

File: web-module/src/main/java/by/tut/mdcatalog/project2/web/security/config/ApiSecurityConfig.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth.userDetailsService(userDetailsService)
+                .passwordEncoder(passwordEncoder);
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.antMatcher("/api/**")
```
[the remark this answers is not in the document]
Comment (s-karpovich) [reply web-module/src/main/java/by/tut/mdcatalog/project2/web/security/config/ApiSecurityConfig.java]: done

File: web-module/src/main/java/by/tut/mdcatalog/project2/web/security/config/WebSecurityConfig.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth.userDetailsService(userDetailsService).passwordEncoder(encoder);
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.authorizeRequests()
+                .antMatchers("/add/**", "/users/**")
```
[the remark this answers is not in the document]
Comment (s-karpovich) [reply web-module/src/main/java/by/tut/mdcatalog/project2/web/security/config/WebSecurityConfig.java]: done

File: web-module/src/main/resources/dbinit.sql [hunk 7]
```diff
@@ -0,0 +1,145 @@
+DROP TABLE IF EXISTS review;
```
[the remark this answers is not in the document]
Comment (s-karpovich) [reply web-module/src/main/resources/dbinit.sql]: done

File: web-module/pom.xml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-security</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.thymeleaf.extras</groupId>
+            <artifactId>thymeleaf-extras-springsecurity5</artifactId>
+            <version>3.0.4.RELEASE</version>
```
[the remark this answers is not in the document]
Comment (s-karpovich) [reply web-module/pom.xml]: done

File: web-module/src/test/java/by/tut/mdcatalog/project2/web/controller/ReviewControllerIntegrationTest.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+    @WithMockUser(authorities = "ADMINISTRATOR")
+    @Test
+    public void shouldShowReviewsPageforAdmin() throws Exception {
+        mvc.perform(get("/reviews"))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    @WithMockUser(authorities = {"ADMINISTRATOR"})
```
[the remark this answers is not in the document]
Comment (s-karpovich) [reply web-module/src/test/java/by/tut/mdcatalog/project2/web/controller/ReviewControllerIntegrationTest.java]: done

File: web-module/src/test/java/by/tut/mdcatalog/project2/web/controller/UserControllerIntegrationTest.java (hunk 1 above)
Comment (s-karpovich) [reply web-module/src/test/java/by/tut/mdcatalog/project2/web/controller/UserControllerIntegrationTest.java]: done

File: service-module/src/main/java/by/tut/mdcatalog/project2/service/model/UserDTO.java [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+    @Size(min = 1, max = 50)
+    private String surname;
+
+    private Boolean isDeleted;
+    @NotNull
+    private String role;
+
+    public UserDTO() {
+    }
+    public UserDTO(
```
[the remark this answers is not in the document]
Comment (s-karpovich) [reply service-module/src/main/java/by/tut/mdcatalog/project2/service/model/UserDTO.java]: done

File: repository-module/src/main/java/by/tut/mdcatalog/project2/repository/impl/ReviewRepositoryImpl.java [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+            logger.error(e.getMessage(), e);
+            throw new DataBaseException(String.format(RepositoryErrors.DATABASE_QUERY_ERROR, sqlQuery), e);
+        }
+    }
+
+    private Review buildReview(ResultSet resultSet) throws SQLException {
+        Review review = new Review();
+        review.setId(resultSet.getLong("id"));
+        try {
+            review.setDate(new SimpleDateFormat("dd/MM/yyyy").parse(resultSet.getString("date")));
```
[the remark this answers is not in the document]
Comment (s-karpovich) [reply repository-module/src/main/java/by/tut/mdcatalog/project2/repository/impl/ReviewRepositoryImpl.java]: done

File: service-module/src/main/java/by/tut/mdcatalog/project2/service/impl/ReviewServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+    @Override
+    public List<ReviewDTO> getReviews() {
+        try (Connection connection = reviewRepository.getConnection()) {
+            try {
+                connection.setAutoCommit(false);
+                List<ReviewDTO> reviewDTOList = new ArrayList<>();
+                List<Review> reviewList = reviewRepository.getReviews(connection);
+                for (Review review : reviewList) {
+                    User user = userRepository.getById(connection, review.getUser().getId());
+                    if (user == null) continue;     // Skip to show reviews of deleted users
```
[the remark this answers is not in the document]
Comment (s-karpovich) [reply service-module/src/main/java/by/tut/mdcatalog/project2/service/impl/ReviewServiceImpl.java]: done

**Pull request merged**